src/main/java/es/thalesalv/gptbot/adapters/discord/listener/AutoCompleteListener.java

```diff
@@ -0,0 +1,32 @@
+package es.thalesalv.gptbot.adapters.discord.listener;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.springframework.stereotype.Service;
+
+import net.dv8tion.jda.api.events.interaction.command.CommandAutoCompleteInteractionEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.interactions.commands.Command;
+
+@Service
+public class AutoCompleteListener extends ListenerAdapter {
+
+    @Override
+    public void onCommandAutoCompleteInteraction(CommandAutoCompleteInteractionEvent event) {
+
+        final String eventName = event.getName();
+        final String focusedOption = event.getFocusedOption().getName();
+        if (eventName.equals("lorebook") && focusedOption.equals("action")) {
+
+            final String[] loreBookOptions = { "create", "retrieve", "update", "delete" };
+            List<Command.Choice> options = Stream.of(loreBookOptions)
+                    .filter(opt -> opt.startsWith(event.getFocusedOption().getValue()))
+                    .map(opt -> new Command.Choice(opt, opt))
```

Comment:
This looks weird to pass in opt twice

Reply:
In that case it's because "create", "retrieve", "update" and "delete" are both the name of the option and the value as well. But this entire listener didn't work as expected. I'll see what I can do to make it work or remove it entirely.